src/config_parser_utils/merge_configuration_files.py

```diff
@@ -15,59 +19,108 @@
-        sys.exit(1)
+    for file_path in (old_file_path, new_file_path):
+        if not os.path.isfile(file_path):
+            logger.error("file %s does not exist.", file_path)
+            return False
 
     # Create a configparser object
     config_old = configparser.ConfigParser()
-    config_new = configparser.ConfigParser()
-    config_merged = configparser.ConfigParser()
-
-    # Read the old and new files
-    try:
-        config_old.read(old_file_path)
-        config_new.read(new_file_path)
+    config_old.optionxform = str  # Preserve case sensitivity
 
-    except configparser.Error as e:
-        logging.error("Failed to parse configurations files: %s", e)
-        sys.exit(1)
+    res = config_old.read(old_file_path)
+    if not res:
+        logger.error("Error: %s exists but could not be read properly.", old_file_path)
+        return False
 
-    # Start with the new configuration as the base for the merged configuration
-    for section in config_new.sections():
-        config_merged.add_section(section)
-        for option in config_new.options(section):
-            config_merged.set(section, option, config_new.get(section, option))
+    try:
+        with open(new_file_path, 'r', encoding='utf-8') as nf, open(merged_file_path, 'w', encoding='utf-8') as of:
+            section = None
+            for line in nf:
+                stripped = line.strip()
 
-    # Override values in the merged configuration with those from the old configuration
-    for section in config_merged.sections():
-        if config_old.has_section(section):
-            for option in config_merged.options(section):
-                if config_old.has_option(section, option):
-                    # Override the value in the merged configuration
-                    config_merged.set(section, option, config_old.get(section, option))
+                # Preserve section headers
+                if stripped.startswith("[") and stripped.endswith("]"):
+                    section = stripped[1:-1]
+                    of.write(line)
+                    continue
 
-    # Write the merged configuration to the old file path
-    with open(old_file_path, 'w', encoding="utf-8") as configfile:
-        config_merged.write(configfile)
+                # Match key-value pairs, preserving inline comments
+                match = re.match(CFG_LINE_RGX, line)
+                if match and section:
+                    key, new_value = match.groups()
+                    if config_old.has_section(section) and config_old.has_option(section, key):
+                        new_value = (config_old.get(section, key)).strip()
+                    # Write back the line with preserved comments
+                    of.write("%s = %s\n" % (key, new_value))
+                else:
+                    of.write(line)  # Preserve non-key lines (empty lines, comments)
+   
+    except Exception as e:
+        logger.error("An unexpected error occurred: %s" % e)
```

Comment:
Also there - Failed to set a value or to process a line ... and then error.
also use f"{variable}"


the UFM script was written for python 2.7 ... so use 3.10 abilities now